File: app/src/main/java/workshop1024/com/xproject/net/XRetrofit.java
package workshop1024.com.xproject.net;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import workshop1024.com.xproject.XConstant;

public class XRetrofit {
    private static Retrofit mRetrofit;

    public static Retrofit getRetrofit() {
        if (mRetrofit == null) {
            mRetrofit = new Retrofit.Builder().baseUrl(XConstant.SERVER_URL).addConverterFactory(GsonConverterFactory.
                    create()).build();
        }
        return mRetrofit;
    }
}

File: app/src/main/java/workshop1024/com/xproject/net/XRetrofit.kt
package workshop1024.com.xproject.net

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import workshop1024.com.xproject.XConstant

object XRetrofit {
    private var mRetrofit: Retrofit? = null

    val retrofit: Retrofit?
        get() {
            if (mRetrofit === null) {
                mRetrofit = Retrofit.Builder().baseUrl(XConstant.SERVER_URL).addConverterFactory(GsonConverterFactory.create()).build()
            }
            return mRetrofit
        }
}